dev/fattest.simplicity/src/componenttest/containers/KeystoreBuilder.java

```diff
@@ -0,0 +1,337 @@
+import java.util.Objects;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.shaded.com.google.common.io.Files;
+
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * A builder classes that will copy certificates and/or keys out of a container
+ * and add them to a keystore file locally for testing.
+ */
+public class KeystoreBuilder {
+
+    /** Standard store types */
+    public static enum STORE_TYPE {
+        JKS(".jks"),
+        PKCS12(".p12");
+
+        public String extension;
+
+        private STORE_TYPE(String extension) {
+            this.extension = extension;
+        }
+    };
+
+    // Required fields for builder
+    private final LibertyServer server;
+    private final GenericContainer<?> container;
+
+    // Required configurations
+    private String trustoreDirectory;
+
+    // Default configurations
+    private String truststoreFilename = "truststore";
+    private String truststorePassword = "liberty";
+    private STORE_TYPE storeType = STORE_TYPE.PKCS12;
+    private final Map<String, MigratingCertificate> certificates = new HashMap<>();
+
+    ///// CONSTRUCTOR /////
+
+    private KeystoreBuilder(LibertyServer server, GenericContainer<?> container) {
+        this.server = server;
+        this.container = container;
+    }
+
+    ///// BUILDER /////
+
+    public static KeystoreBuilder of(LibertyServer server, GenericContainer<?> container) {
+        validateState(server, container);
+        return new KeystoreBuilder(server, container);
+    }
+
+    ///// CONFIGURATIONS /////
+
+    /**
+     * A certificate that exists in the container.
+     * This can be called multiple times to add more certificates.
+     *
+     * @param  alias  the certificate alias
+     * @param  source the fully qualified path to the certificate in the container
+     *
+     * @return        this
+     */
+    public KeystoreBuilder withCertificate(String alias, String source) {
+        certificates.put(alias, MigratingCertificate.of(source));
+        return this;
+    }
+
+    /**
+     * A certificate that exists in the container.
+     * This can be called multiple times to add more certificates.
+     *
+     * @param  alias       the certificate alias
+     * @param  source      the fully qualified path to the certificate in the container
+     * @param  destination the fully qualified path to a client location where the certificate will be copied
+     *
+     * @return             this
+     */
+    public KeystoreBuilder withCertificate(String alias, String source, String destination) {
+        certificates.put(alias, MigratingCertificate.of(source, destination));
+        return this;
+    }
+
+    /**
+     * The directory into which the keystore will be exported.
+     *
+     * @param  directory fully qualified directory path
+     *
+     * @return           this
+     */
+    public KeystoreBuilder withDirectory(String directory) {
+        Objects.nonNull(directory);
```

Comment:
Objects.nonNull returns a boolean, I'm assuming you meant to handle the directory being null here?